### PR TITLE
WIP: add a basic makefile

### PR DESCRIPTION
**Pull request by sp4ke**
Included commands:

- deps: install glide and dependencies
- install: install lnd
- fmt: run `go fmt` excluding vendor
- test: run `go test` as per documentation

### CODE REVIEW

**Comment by Roasbeef** [comment]
Hmm, we don't really use makefile. IMO the go tooling is easy enough to use directly. 

Have you seen the `gotest.sh` script?

**Comment by sp4ke** [comment]
It's just an extra for those who are used working in Unix environments. Usually important github projects include a Makefile which self documents how to build and run the project, and yes I saw the gotest.sh script, it looks a bit messy and unreadable compared to a simple Makefile.

You can ignore my PR if you think it's not needed, you have more important things to focus on :)

**Comment by Roasbeef** [comment]
Thanks for the PR, but this isn't something we're looking to accept at this moment. Typically Go projects don't use makefiles at all, and all the functionality within the makefile is already replicated within the built in tooling, and our own `gotest.sh` script. With that said, a mergeable contribution would possibly be adding some documentation around the test script itself.

**Comment by Roasbeef** [comment]
Discussed this a bit more and we decided this is actually a cool feature! One thing we can do is use the makefil to embed the commit that we're building with _into_ the binary itself.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+deps: glide
+
+install: deps
+	go install . ./cmd/...
+
+fmt:
+	go fmt ./... $$(go list ./... | grep -v '/vendor/')
+
+test:
```
Comment (cfromknecht) [inline Makefile]: Do you have a preference for `make test` vs `make check`? FWIW, this is only running the unit tests, the integration tests can be run using `go test -test.v -tags rpctest -logoutput` on the condition that the latest binary is already installed.

Comment (cfromknecht) [reply Makefile]: Another alternative is that `make test` just runs the unit tests, and `make check` could run the itests after the unit tests succeed.

**Comment by sp4ke** [comment]
Sorry didn't catch the notifications this was reopened. I will make an update soon to this PR based on your comments.

>Do you have a preference for make test vs make check? FWIW, this is only running the unit tests, the integration tests can be run using go test -test.v -tags rpctest -logoutput on the condition that the latest binary is already installed.

>Another alternative is that make test just runs the unit tests, and make check could run the itests after the unit tests succeed.

I am not very familiar with the dev workflow on lnd but I would choose your alternative with both `test` and `check` since tests should run more often during dev .

**Comment by cfromknecht** [comment]
@sp4ke perfect, ping me when I should take another look

**Comment by Roasbeef** [comment]
Superseded by #962.

**Pull request closed without merging**